tests/importer.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::{HashMap, HashSet};
+use std::fs;
+
+use hurban_selector::importer;
+use hurban_selector::importer::{Importer, ImporterError};
+
+fn import_obj(path: &str) -> Vec<importer::Model> {
+    let file_contents = fs::read_to_string(&path).expect("File should be read to string");
+    let (tobj_models, _) =
+        importer::obj_buf_into_tobj(&file_contents).expect("Obj should be parsed");
+
+    importer::tobj_to_internal(tobj_models)
+}
+
+#[test]
+fn test_caches_valid_obj_file() {
+    let mut importer = Importer::new();
+    let path = "tests/fixtures/valid.obj".to_string();
+    let file_contents = fs::read_to_string(&path).expect("File should be read to string");
+    let checksum = importer::calculate_checksum(&file_contents);
+
+    importer
+        .import_obj(&path)
+        .expect("Valid object should be loaded");
+
+    let mut expected_path_checksums = HashMap::new();
+    expected_path_checksums.insert(path.clone(), checksum);
+
+    let mut expected_checksum_paths = HashMap::new();
+    expected_checksum_paths.insert(checksum, vec![path.clone()]);
+
+    let loaded_models_paths: HashSet<_> = importer.loaded_models.keys().collect();
+    let mut expected_loaded_models_paths: HashSet<&String> = HashSet::new();
+    expected_loaded_models_paths.insert(&path);
+
+    assert_eq!(importer.path_checksums, expected_path_checksums);
+    assert_eq!(importer.checksum_paths, expected_checksum_paths);
+    assert_eq!(loaded_models_paths, expected_loaded_models_paths);
+}
+
+#[test]
+fn test_returns_valid_obj_file() {

```

Comment:
This seems to me like it's testing `1 == 1` scenario. Does make sense to have tests like these?

Reply:
I think so. Returning of parsed models is part of that method and we want to be sure it's returning correct ones and not something from the neighborhood.